src/main/java/ch/ethz/matsim/baseline_scenario/additional_traffic/freight/FreightTraffic.java

```diff
@@ -0,0 +1,37 @@
+package ch.ethz.matsim.baseline_scenario.additional_traffic.freight;
+
+import org.matsim.api.core.v01.population.Person;
+import org.matsim.api.core.v01.population.Population;
+import org.matsim.facilities.ActivityFacilities;
+import org.matsim.facilities.ActivityFacility;
+
+public class FreightTraffic {
+    private Population population;
+    private ActivityFacilities facilities;
+
+    public FreightTraffic(Population population, ActivityFacilities facilities) {
+        this.population = population;
+        this.facilities = facilities;
+    }
+
+    public Population getPopulation() {
+        return population;
+    }
+
+    public ActivityFacilities getFacilities() {
+        return facilities;
+    }
+
+    public FreightTraffic add(FreightTraffic freightTraffic) {
```

Comment:
Why fluent interface here?

Reply:
All the stages in the scenario synthesis have the form

`new MyStage(unmodified dependencies).run(modified objects);`

So ideally there should be something like this somewhere:

`new FreightTrafficGenerator(freightData).run(population, facilities);`

Reply:
And so far as I see the whole freight generation is not called anywhere? We can have some "Run*" scripts for debugging or so if it makes sense, but ideally everything should be in the main Run script, and then you have to think how you do the downscaling... do you do it here or will you just use the standard downscaling of the whole population later on?